app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @user = @post.author
+    @comment = @post.comments.build(comment_params)
+    @comment.user = current_user
+
+    if @comment.save
+      flash[:success] = 'Comment created successfully.'
+      redirect_to user_post_path(@user, @post)
+    else
+      flash[:error] = 'Comment could not be created.'
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+
+  def new
+    @comment = Comment.new
+  end
```

Comment:
- [ ] Great work adding the new method as instructed by the previous reviewer. However, This method should not be a private method. It seems you have added it at the bottom of the files among the private methods. Kindly remove it from here and place it at the top. Remember this is a method that is used by both the routes and the views. So it can not be private to this class. 
 - [ ] When you are creating a new comment, it should first find the post using the `params.` So this should be something like, 
`@comment = Post.find_by(id: params[post_id].comment.new`
or
`@comment = post.comments.new`